hackathon_site/dashboard/frontend/src/components/general/Footer/Footer.test.js

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+import Footer from "./Footer";
+
+describe("Footer", () => {
+    test("renders", () => {
+        const { asFragment } = render(<Footer />);
+        expect(asFragment()).toMatchSnapshot();
+    });
+
+    test("date", () => {
+        const fakeDate = new Date("2021-05-17");
+        const fakeYear = "2021";
+
+        const fakeDateClass = class extends Date {
+            constructor() {
+                super(fakeDate);
+            }
+
+            static now() {
+                return null;
+            }
+
+            getFakeYear() {
+                return fakeYear;
+            }
+        };
+
+        global.Date = fakeDateClass;
+        const { getByText } = render(<Footer />);
+        const footerText = getByText(/built/, { exact: false });
+
+        expect(footerText).toHaveTextContent(
+            `© 2020 - ${fakeYear} built by the web team at MakeUofT (University of Toronto)`
```

Comment:
What if the current year is 2020? There should be a test for that as well that the copright just has 2020.

Reply:
If I were to add a mock for the date function would I still need the 2 tests

Reply:
Indeed, you should have two separate tests for this behaviour, because there are two cases: When the year is 2020 (and the copyright just has 2020) and when the year is not 2020 (and the copyright has 2020 - <year>)

Reply:
Implicitly, if you mock the year as 2020 in your first snapshot test, that covers the first case (the second is covered in this test). I would still recommend writing a separate test to be explicit about what's being tested, but that's up to you since it would be redundant.